src/js/admin.js

```diff
@@ -0,0 +1,579 @@
+  changeObjectProperty(id, surname, allTeachers)
+  changeObjectProperty(id, subjectId, allTeachers)
+  localStorage.setItem('teachers', JSON.stringify(allTeachers))
+}
+function editSubject (id, name, year) {
+  const allSubjects = JSON.parse(localStorage.getItem('subjects'))
+  changeObjectProperty(id, name, allSubjects)
+  changeObjectProperty(id, year, allSubjects)
+  localStorage.setItem('subjects', JSON.stringify(allSubjects))
+}
+// === DELETE FUNCTIONS ====
+
+function deleteStudent (id) {
+  const confirmation = confirm('Are you sure?')
+  if (confirmation) {
+    const allStudents = JSON.parse(localStorage.getItem('students'))
+    const allParents = JSON.parse(localStorage.getItem('parents'))
+    const allGrades = JSON.parse(localStorage.getItem('grades'))
+    const parentId = allStudents.filter((student) => student.id === id)[0]
+      .parentId
+
+    const studentsWithoutThisStudent = allStudents.filter(
+      (student) => student.id !== id
+    )
+    const parentsWithoutThisParent = allParents.filter(
+      (parent) => parent.id !== parentId
+    )
+    const gradesWithoutThisStudentsGrades = allGrades.filter(
+      (grade) => grade.student !== id
+    )
+    localStorage.setItem(
+      'students',
+      JSON.stringify(studentsWithoutThisStudent)
+    )
+
+    localStorage.setItem('parents', JSON.stringify(parentsWithoutThisParent))
+    localStorage.setItem(
+      'grades',
+      JSON.stringify(gradesWithoutThisStudentsGrades)
+    )
+    createDisplayForAllStudents()
+  }
+}
+function deleteTeacher (id) {
+  const confirmation = confirm('Are you sure?')
+  if (confirmation) {
+    const allTeachers = JSON.parse(localStorage.getItem('teachers'))
+    const teachersWithoutThisTeacher = allTeachers.filter(
+      (teacher) => teacher.id !== id
+    )
+    localStorage.setItem(
+      'teachers',
+      JSON.stringify(teachersWithoutThisTeacher)
+    )
+    createDisplayForAllTeachers()
+  }
+}
+function deleteSubject (id) {
+  const confirmation = confirm('Are you sure?')
+  if (confirmation) {
+    const allSubjects = JSON.parse(localStorage.getItem('subjects'))
+    const subjectsWithoutThisSubject = allSubjects.filter(
+      (subject) => subject.id !== id
+    )
+    localStorage.setItem(
+      'subjects',
+      JSON.stringify(subjectsWithoutThisSubject)
+    )
+    createDisplayForAllSubjects()
+  }
+}
+// === DOM FUNCTIONS ===
+
+function createDisplayForAllStudents () {
+  const dataFromStorage = d('mainMenu')
+  const allstudents = JSON.parse(localStorage.getItem('students'))
+  let output = `<table>
+  <thead>
+    <tr>
+      <th>ID</th>
+      <th>Name</th>
+      <th>Surname</th>
+      <th></th>
+      <th></th>
+    </tr>
+  </thead>`
+  for (const person of allstudents) {
+    output += `<tr>
+        <td>${person.id}</td><td>${person.name}</td>
+        <td>${person.surname}</td>
+        <td><button data-id="${person.id}" onclick="createDisplayForStudentUpdate(${person.id})">Edit</button></td>
```

Comment:
Probaj da zameniš ove onclick atribute sa event listenerima.

Reply:
Probao sam ali jako tesko ide. Prvo, funkcija unutar eventListenera ne prima parametre, ali to sam i nekako resio sa anonimnom funkcijom koja poziva imenovanu tipa d('editStudentButton').addEventListener('click',function(){ createDisplayForStudentUpdate(person.id)})  .. ali onda nastaje nova muka jer ovaj editStudentButton zapravo ne moze da bude id (jer ih ima onoliko koliko ima ucenika) , nego mora da bude class. Na sve to mora da se ponovo doda script tag da bi se to sve unelo jer se ucitava dinamicki, pa u inicijalnom js fajlu d('editStudentButton') vraca undefined. Pretpostavljam da postoji lakse resenje, ali ja ne mogu da ga provalim.

Reply:
Dodaješ eventListener na body ili na nešto što je malo bliže tome i ne menja se dinamički
onda proveriš npr. if(event.target.dataset.studentId){
// i onda ovde kod ako imaš data-student-id
}

Reply:
A za taj dinamički skript mislim da je problem samo u redosledu, ne bih smeo da se kladim